Link failure from using #c_varargs parameter as an import

## Context

```
        Odin:    dev-2026-06:70607eeb0
        OS:      Windows 11 Home Basic (version: 25H2), build 26200.8655
        CPU:     AMD Ryzen AI 7 350 w/ Radeon 860M
        RAM:     15525 MiB
        Backend: LLVM 20.1.0
```

## Current Behavior

Currently, the linking step fails with this output when running the reproducible example
```
$ ./odin run main.odin -file
main-main-142ef031370.obj : error LNK2019: unresolved external symbol args referenced in function main::main.cfoo-0
C:\Users\REDACTED\Repos\tf2spi\Odin\main.exe : fatal error LNK1120: 1 unresolved externals
```

## Expected Behavior

I expect the example to compile and output the following
```
[1, 2, 3, 4]
Hello World!
```

## Failure Information (for bugs)

While trying to test #6908, I ran into this error when trying to use C varargs

### Steps to Reproduce

The reproducible example is below in ``main.odin``

```odin
package main
import "core:fmt"
import "base:runtime"

main :: proc() {
        cfoo :: proc "c" (#c_vararg args: ..i8) { context = runtime.default_context(); fmt.printf("%v\n", args) }
        foo :: proc(array: [$N] i8) { cfoo(expand_values(array)) }
        foo([4]i8{1, 2, 3, 4})
        fmt.println("Hello World!")
}
```

Run ``odin run main.odin -file`` to reproduce

--- a/src/check_builtin.cpp
+++ b/src/check_builtin.cpp
@@ -765,8 +765,10 @@ gb_internal bool check_builtin_c_procedure(CheckerContext *c, Operand *operand, 
 			return false;
 		}
 
+		c->allow_c_vararg_param = true;
 		Operand args = {};
 		check_expr(c, &args, ce->args[1]);
+		c->allow_c_vararg_param = false;
 		if (list.mode == Addressing_Invalid) {
 			return false;
 		}

--- a/src/check_expr.cpp
+++ b/src/check_expr.cpp
@@ -1968,6 +1968,14 @@ gb_internal Entity *check_ident(CheckerContext *c, Operand *o, Ast *n, Type *nam
 		break;
 
 	case Entity_Variable:
+		if ((e->flags & EntityFlag_CVarArg) && !c->allow_c_vararg_param) {
+			ERROR_BLOCK();
+			error(o->expr, "'#c_vararg' parameter '%.*s' cannot be used directly", LIT(name));
+			error_line("\tSuggestion: use c_va_start to convert C varargs to c_va_list\n");
+			o->mode = Addressing_Invalid;
+			o->type = t_invalid;
+			return e;
+		}
 		e->flags |= EntityFlag_Used;
 		if (type == t_invalid) {
 			o->type = t_invalid;

--- a/src/checker.hpp
+++ b/src/checker.hpp
@@ -834,6 +834,7 @@ struct CheckerContext {
 	bool       hide_polymorphic_errors;
 	bool       in_polymorphic_specialization;
 	bool       allow_arrow_right_selector_expr;
+	bool       allow_c_vararg_param;
 	u8         bit_field_bit_size;
 	Scope *    polymorphic_scope;
 

